src/main/java/uk/gov/hmcts/reform/blobrouter/reconciliation/controller/ReconciliationReportGenerationController.java

```diff
@@ -0,0 +1,54 @@
+package uk.gov.hmcts.reform.blobrouter.reconciliation.controller;
+
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.format.annotation.DateTimeFormat;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import uk.gov.hmcts.reform.blobrouter.config.TargetStorageAccount;
+import uk.gov.hmcts.reform.blobrouter.reconciliation.service.DetailedReportService;
+import uk.gov.hmcts.reform.blobrouter.reconciliation.service.ReconciliationMailService;
+import uk.gov.hmcts.reform.blobrouter.reconciliation.service.SummaryReportService;
+
+import java.time.LocalDate;
+import java.util.Arrays;
+import java.util.List;
+
+import static org.springframework.format.annotation.DateTimeFormat.ISO.DATE;
+
+@RestController
+@RequestMapping(path = "/reconciliation-reports")
+@ConditionalOnProperty("reconciliation.manual-report-generation-and-email-enabled")
+public class ReconciliationReportGenerationController {
+
+    private final ReconciliationMailService reconciliationMailService;
+    private final DetailedReportService detailedReportService;
+    private final SummaryReportService summaryReportService;
+
+    public ReconciliationReportGenerationController(
+        ReconciliationMailService reconciliationMailService,
+        DetailedReportService detailedReportService,
+        SummaryReportService summaryReportService
+    ) {
+        this.reconciliationMailService = reconciliationMailService;
+        this.detailedReportService = detailedReportService;
+        this.summaryReportService = summaryReportService;
+    }
+
+    @PostMapping(path = "/generate-and-email-report")
+    public void generateAndEmailReports(
+        @RequestParam(name = "date") @DateTimeFormat(iso = DATE) LocalDate date
+    ) {
+        List<TargetStorageAccount> availableAccounts = Arrays.asList(
+            TargetStorageAccount.CFT, TargetStorageAccount.CRIME
+        );
```

Comment:
why not all accounts?

Reply:
In the existing `ReconciliationMailTask` class, only CFT and CRIME are used. shall I change it to all accounts?

Reply:
Updated to generate reports for all storage accounts